### DISCUSSION TITLE
Bar chart onclick event

### DISCUSSION THREAD

**Discussion by kekecr** [vikramlearning/blazorbootstrap#461] [Q&A]
Is it possible to add an onclick event to bar chart so that when you click on a bar new data is fetched form the server based on the bar index you clicked on

**Comment by gvreddy04** [comment]
@kekecr Thank you for using BlazorBootstrap! Could you share a real-time scenario where you would want to implement a click event?

**Comment by kekecr** [comment]
like here https://www.wapmat.com/, the auction chart, I have it set up with apache echarts, so when you clcik on a bar a new data set is loaded. I decided to rewrite the code with bootstrap blazor and was wondering if i can make the same functionality with blazor bootstrap bar chart. And also if it is possible to add zoom slider window to bar chart like on the echart on above site, makes it easier to clik on a bar.

**Comment by JamInnoBox** [comment]
Onclick event on the bar chart would be much appreciated. I have a chart where i have a sub page for each bar. I want to be able to navigate to the sub page when clicking on the bar.

**Comment by rtiwari-se** [comment]
@gvreddy04  can you please share update on this, trying for pie chart onclick event. when clicked on pie chart want to know which label being clicked.

**Comment by BIMEnhanced** [comment]
Hello,
I found out that the onclick functionnality exists [(See this commit)](https://github.com/vikramlearning/blazorbootstrap/pull/961).
It seem to have been merged to main but I can't find the actual code in the repos.
Am I missing something here ?
Thank for the help.

**Reply by gvreddy04** [reply]
@BIMEnhanced Not yet merged. This will be released in the next release.

**Reply by stalawina** [reply]
@gvreddy04 I just noticed a new release 3.4.0 . Do you know if onclick events are now possible with a bar chart?
Thanks!

**Reply by gvreddy04** [reply]
@stalawina The OnClick feature will be released with version 3.4.1 later this month. We are now planning to push updates more frequently.

**Reply by stalawina** [reply]
@gvreddy04 I noticed we are now on 3.5.0, but I am not able to find a OnClick event feature. Do you have a documentation for it?
